src/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.cxx

```diff
@@ -168,21 +201,39 @@ get_output(DiscretisedDensity<3,float> &density) const
                          p.get_const_data_ptr() + offset,
                          num_lors_per_chunk,
                          _helper->imgdim.data(),
-                         /*threadsperblock*/ 64);
-
+                         /*threadsperblock*/ 64
+                         );
+      }
       offset += num_lors_per_chunk;
     }
 
     // sum backprojected images on the first CUDA device
-    sum_float_arrays_on_first_device(image_on_cuda_devices, num_image_voxel);
+    sum_float_arrays_on_first_device(image_on_cuda_devices, _helper->num_image_voxel);
 
     // copy summed image back to host
-    get_float_array_from_device(image_on_cuda_devices, num_image_voxel, 0, image_vec.data());
+    get_float_array_from_device(image_on_cuda_devices, _helper->num_image_voxel, 0, image_vec.data());
 
     // free image array from CUDA devices
     free_float_array_on_all_devices(image_on_cuda_devices);
 
 #else
+    if (this->_proj_data_info_sptr->is_tof_data() == 1)
+      {
+        std::vector<float> mem_for_PP_back(_helper->num_lors * _helper->num_tof_bins);
+        const float* STIR_mem = p.get_const_data_ptr();
+
+        TOF_transpose(mem_for_PP_back, STIR_mem, _helper, offset);
+
+        joseph3d_back_tof_sino(_helper->xend.data(), _helper->xstart.data(), image_vec.data(), _helper->origin.data(),
+                               _helper->voxsize.data(), mem_for_PP_back, num_lors, _helper->imgdim.data(), tofbin_width,
+                               &sigma_tof, &tofcenter_offset,
+                               4, // float n_sigmas,
+                               _projected_data_sptr->get_proj_data_info_sptr()->get_num_tof_poss(),
+                               0, //  unsigned char lor_dependent_sigma_tof
+                               0  // unsigned char lor_dependent_tofcenter_offset
+        );
+      }
```

Comment:
Is this code duplication from `#if parallelproj_built_with_CUDA`

Reply:
yes, but only almost. We don't do chunking here. (We could maybe, to avoid creating a lot of memory).

However, I'm asking about transpose on GPU, so let's postpone that.